beeline/src/java/org/apache/hive/beeline/Rows.java

```diff
@@ -51,14 +55,14 @@ abstract class Rows implements Iterator {
     this.beeLine = beeLine;
     nullStr = beeLine.getOpts().getNullString();
     rsMeta = rs.getMetaData();
-    int count = rsMeta.getColumnCount();
-    primaryKeys = new Boolean[count];
     if (beeLine.getOpts().getNumberFormat().equals("default")) {
       numberFormat = null;
     } else {
       numberFormat = new DecimalFormat(beeLine.getOpts().getNumberFormat());
     }
     this.convertBinaryArrayToString = beeLine.getOpts().getConvertBinaryArrayToString();
+
+    primaryKeys = initializePrimaryKeyMetadata();
```

Comment:
IIUC primary key info is queried every time a new `Rows` instance is created. Even if coloring is not needed. Could you please explore if this can be improved?